lib/src/path_observer.dart

```diff
@@ -290,10 +290,10 @@ _getObjectProperty(object, property) {
     // whether the type practically implements the indexer API
     // (smoke.hasInstanceMethod(type, const Symbol('[]')))?
     if (object is Indexable || object is Map && !_MAP_PROPERTIES.contains(property)) {
-      return object[smoke.symbolToName(property)];
+      return object[property];
     }
     try {
-      return smoke.read(object, property);
+      return jsProxyReflectable.reflect(object).invokeGetter(property);
```

Comment:
Hmm this is really a general `reflectable` design issue. There isn't one global mirror system, each annotation which extends `Reflectable` effectively establishes its own mirror system so you don't know which one to use. With the current design the best way to deal with this would probably be to accept a reflectable instance as an argument to the constructor, and that would be used to do all reflection? You could then extend this class with a custom implementation which automatically passes the right annotation through the super constructor so your users don't have to see that detail?

Another (probably better) long term option would be to make a feature request to reflectable which would allow you to reflect on an arbitrary class and just ask for certain capabilities, and it would automatically use any available reflectable annotation which provides those (or fail if none were available).